### DISCUSSION TITLE
how to save/serialize trained sktime model

### DISCUSSION THREAD

**Discussion by fkiraly** [alan-turing-institute/sktime#2006] [Q&A]
adding some questions from Gitter here:

from @oda356:
"Hi, I can't find an API to save/serialize my trained model in sktime document, how can I save my trained model so that I needn't train my model again when I want to predict some data?"

**Comment by fkiraly** [comment]
@oda356, the "officially" supported way would be using pickle, i.e,. using `pickle.dumps` and `pickle.loads`.

**Reply by BradKML** [reply]
Side question: how would one run a model on Streamlit? I have seen some some code below for running it in Colab. Their app.py would have to contain the loaded pickled models. However trying to create a model-building service in Streamlit (supplying data and getting forecast) would be an issue with Colab's limited resources.
```
!npm install localtunnel
!cd /content
!streamlit run /content/app.py &>/content/logs.txt &
!npx localtunnel --port 8501

**Reply by topher-lo** [reply]
@BrandonKMLee If you want a serialization method that is more robust and where your pickle can be deserialized in different environments, use `cloudpickle`.

**Reply by topher-lo** [reply]
Also I think you should post your streamlit question on streamlit.
